src/lucky_record/associations.cr

```diff
@@ -72,7 +72,14 @@ module LuckyRecord::Associations
               end
               .preload_{{ through.id }}
               .distinct
-            {{ assoc_name }} = {} of Int32 => Array({{ model }})
+
+            {% owner_id_type = if model.id == "Product"
```

Comment:
So yeah @paulcsmith I'm currently stuck resolving this conditional, which should obviously be `    {% owner_id_type = model.resolve.constant(:PRIMARY_KEY_TYPE_CLASS) %}`, however that fails in a way I don't understand:

```
Error in line 1: while requiring "./spec/insert_spec.cr"

in spec/insert_spec.cr:1: while requiring "./spec_helper"

require "./spec_helper"
^

in spec/spec_helper.cr:3: while requiring "./support/**"

require "./support/**"
^

in spec/support/comment.cr:1: while requiring "./post"

require "./post"
^

in spec/support/post.cr:9: expanding macro

    has_many tags : Tag, through: :taggings
    ^~~~~~~~

in src/lucky_record/associations.cr:76: undefined constant Tag

            {% owner_id_type = model.resolve.constant(:PRIMARY_KEY_TYPE_CLASS) %}
                                     ^~~~~~~

in spec/support/post.cr:9: undefined constant Tag

    has_many tags : Tag, through: :taggings
```

Why does that suddenly not know the Tag class? I tried `require`-ing the models error by error but ultimately couldn't fix the undefined constants errors. Any ideas?

Reply:
That is definitely strange. I'll have to pull it down and check it out!